zerver/views/report.py

```diff
@@ -61,26 +62,26 @@ def report_send_times(request: HttpRequest, user_profile: UserProfile,
         statsd.incr('render_disparity')
     return json_success()
 
-@human_users_only
 @has_request_variables
-def report_narrow_times(request: HttpRequest, user_profile: UserProfile,
+def report_narrow_times(request: HttpRequest, user_profile: Optional[UserProfile],
                         initial_core: int=REQ(converter=to_non_negative_int),
                         initial_free: int=REQ(converter=to_non_negative_int),
                         network: int=REQ(converter=to_non_negative_int)) -> HttpResponse:
     request._log_data["extra"] = f"[{initial_core}ms/{initial_free}ms/{network}ms]"
-    base_key = statsd_key(user_profile.realm.string_id, clean_periods=True)
+    realm = get_valid_realm_from_request(request)
+    base_key = statsd_key(realm.string_id, clean_periods=True)
     statsd.timing(f"narrow.initial_core.{base_key}", initial_core)
     statsd.timing(f"narrow.initial_free.{base_key}", initial_free)
     statsd.timing(f"narrow.network.{base_key}", network)
     return json_success()
 
-@human_users_only
 @has_request_variables
-def report_unnarrow_times(request: HttpRequest, user_profile: UserProfile,
+def report_unnarrow_times(request: HttpRequest, user_profile: Optional[UserProfile],
```

Comment:
This is a `Union[UserProfile, AnonymousUser]`, right?

Reply:
yeah, updated.